### ISSUE TITLE
RuntimeError in running bean-example

### ISSUE THREAD

**Issue by blais** [beancount/beancount#322]
**[Original report](https://bitbucket.org/blais/beancount/issue/322) by Mike Myat Min Han (Bitbucket: Mike Han, GitHub: [mmhan](https://github.com/mmhan)).**

----------------------------------------

I'm running it for the first time, after installing it via pip3 like `sudo pip3 install beancount
`

I'm on MacOSX. Let me know if you need any other info.

```
INFO    : Generating Salary Employment Income
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/beancount/utils/misc_utils.py", line 143, in skipiter
    value = next(sit)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/bean-example", line 4, in <module>
    from beancount.scripts.example import main; main()
  File "/usr/local/lib/python3.7/site-packages/beancount/scripts/example.py", line 1777, in main
    file=output_file)
  File "/usr/local/lib/python3.7/site-packages/beancount/scripts/example.py", line 1541, in write_example_file
    date_begin, date_end)
  File "/usr/local/lib/python3.7/site-packages/beancount/scripts/example.py", line 459, in generate_employment_income
    dtstart=date_begin, until=date_end), 2):
RuntimeError: generator raised StopIteration
```

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Are you using Python 3.7?

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #322: Adjustments to generator's support of StopIteration.
